servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ChannelSet.java

```diff
@@ -87,14 +97,17 @@ public ChannelSet(Executor offloadingExecutor) {
      */
     public boolean addIfAbsent(final Channel channel) {
         final boolean added = channelMap.putIfAbsent(channel.id(), channel) == null;
+        if (added) {
+            final ConnectionObserver observer = this.connectionObserverFactory != null ?
+                    this.connectionObserverFactory.get() : null;
+            if (observer != null) {
+                assignConnectionObserver(channel, observer);
```

Comment:
its somewhat awkward to handle the `ConnectionObserver` in the `ChannelSet` on the server, and directly in the transport code for the client. did you consider moving this into the transport as well?